### DISCUSSION TITLE
Is it a Cubic B-Spline Function?

### DISCUSSION THREAD

**Discussion by swaraj1112** [ttk592/spline#14] [Q&A] [answered]
I ran the code twice with 5 control points, in the second run I changed the fourth control point by a certain magnitude and observed that the global shape of the spline changed completely. I was hoping for more local control.

**Comment by ttk592** [comment]
I don't use b-splines as a basis internally, but the default spline (`tk::spline::cspline`) will create the same result as if the interpolation was done with cubic b-splines as basis. See `examples/plot_bspline.cpp`. However, why do you expect a local behaviour with b-splines? If you modify a single grid point (x<sub>i</sub>, y<sub>i</sub>) it will modify all the coefficients and the entire spline will change globally. For a more local behaviour you can use `tk::spline s(X,Y,tk::spline::cspline_hermite)` but this is only once cont differentiable and therefore not a b-spline.

Results for b-splines and `tk::spline::cspline` are indistinguishable:
```
$ ./plot_bspline.sh -b 1 -t cspline
```
![bspline](https://user-images.githubusercontent.com/1814760/108894516-ac788300-7612-11eb-8c5d-79e2a6b0a073.gif)

```
$ ./plot_bspline.sh -b 1 0 0 0 1 0 0.5 -t cspline
```
![bspline2](https://user-images.githubusercontent.com/1814760/108894846-20b32680-7613-11eb-89ec-dac99d1cb90a.gif)

**Comment by ttk592** [comment, marked as answer]
To illustrate, changing a single input grid point has the following impact:

C<sup>2</sup> spline (`tk::spline::cspline`): impacts the entire spline
![c2spline](https://kluge.in-chemnitz.de/opensource/spline/cubic_c2_spline_anim.gif)

C<sup>1</sup> Hermite spline (`tk::spline::cspline_hermite`): impacts two adjacent segments on each side
![c1hermitespline](https://kluge.in-chemnitz.de/opensource/spline/cubic_hermite_spline_anim.gif)
